### DISCUSSION TITLE
help; searxng web search 400 clinet error: Bad request

### DISCUSSION THREAD

**Discussion by qhmhl** [open-webui/open-webui#2948] [General]
# Bug Report

## Description
web search with searxng, feedback error "400 clinet error: Bad request"

**Bug Summary:**
[Provide a brief but clear summary of the bug]
400 Client Error: BAD REQUEST for url: http://192.168.3.6:38080/search?q=&q=%E5%B8%AE%E6%88%91%E6%9F%A5%E8%AF%A22024%E4%B8%AD%E5%9B%BD%E9%AB%98%E8%80%83%E5%85%B7%E4%BD%93%E6%97%B6%E9%97%B4&format=json&pageno=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0

**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]

**Expected Behavior:**
[Describe what you expected to happen.]

**Actual Behavior:**
[Describe what actually happened.]

## Environment

- **Open WebUI Version:** [e.g., 0.1.120]
- **Ollama (if applicable):** [e.g., 0.1.30, 0.1.32-rc1]

- **Operating System:** [e.g., Windows 10, macOS Big Sur, Ubuntu 20.04]
- **Browser (if applicable):** [e.g., Chrome 100.0, Firefox 98.0]

## Reproduction Details

**Confirmation:**

- [ ] I have read and followed all the instructions provided in the README.md.
- [ ] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

[Describe the method you used to install the project, e.g., manual installation, Docker, package manager, etc.]

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by j-chacko** [comment]
### Description
Open-WebUI, Web Search feature using SearXNG returns an error:
`403 Client Error: FORBIDDEN for url: http://<PUBLIC_IP_SEARXNG_SERVER>/search?q=Adele&format=json&pageno=1&safesearch=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0`

**Steps to reproduce:**
1. Install SearXNG on its own instance. I followed the installation method using the [install script](https://github.com/goshawk22/searxng-installation/tree/main?tab=readme-ov-file#starting-the-installation) and not the docker method. I had some issues with accessing SearXNG using my Public IP Address, so I followed [this guide](https://github.com/goshawk22/searxng-installation/tree/main) to set up an Nginx Reverse Proxy. I can access SearXNG through my browser, which returns search results for any query.
2. Install Ollama on a GPU instance following the default install method for Linux from [ollama.com](https://ollama.com/download/linux). I changed the Environment variable in `/etc/systemd/system/llama.service` so that I can access the API from a Public IP.
3. Install Open-WebGUI by following the official directions to install Open-WEBUI in docker when Ollama is hosted on a different server `docker run -d -p 3000:8080 -e OLLAMA_BASE_URL=https://example.com -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main`
4. Now, I can access Ollama models through Open-WebUI and chat with the LLM model I pulled.
5. In the Admin Panel for Open-WebUI > Web Search > Enable Web Search > Web Search Engine: `searxng` > Searxng Query URL: `http://<PUBLIC_IP_SEARXNG_SERVER>/search?q=<query>`
6. Open a new chat > in the chat box `+` and enable Web Search, type text `Adele` in the conversation box and send to the LLM

**Expected Behavior:**
I am unsure of the expected behaviour as I am trying it out for the first time. But I guess I should get some results back from the search query and not an error.

**Actual Behavior:**
Returns the error `403 Client Error: FORBIDDEN for url: http://<PUBLIC_IP_SEARXNG_SERVER>/search?q=Adele&format=json&pageno=1&safesearch=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0`

**Environment**
- Ollama version is 0.1.44. Using the latest version of the model's Phi and Llama3:8b
- Searxng 2024.6.14+7be468d21
- Open-WebUI 0.3.4
- Browser: I have tried the latest versions of Edge, Chrome and Firefox

**Additional Information**
- If I take the URL generated and send to SearXNG and enter it directly into the browser address field, i.e. `http://<PUBLIC_IP_SEARXNG_SERVER>/search?q=Adele&format=json&pageno=1&safesearch=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0`, I get the error `Forbidden
You don't have the permission to access the requested resource. It is either read-protected or not readable by the server.`
- Removing each of the flags added to the URL from the end, 
  - Remove `&image_proxy=0`; forbidden error
  - Remove `&theme=simple'; forbidden error
  - Remove `&categories=`; forbidden error
  - Remove `/&time_range=`; too many requests error
  - Remove `&language=en-US`; too many requests error
  - Remove `&safesearch=1`; too many requests error
  - Remove `&pageno=1`; too many requests error
  - Remove `&format=json`; too many requests error
  - When I only send `http://<PUBLIC_IP_SEARXNG_SERVER>/search?q=Adele`; I successfully get search results

**Reply by 4R9UN** [reply]
error due to "&format=json&" in URL

**Reply by ips972** [reply]
no need, just make sure to add json in the response that the system returns by default:
look at settings.yml make sure it looks like this:


   remove format to deny access, use lower case.
   formats: [html, csv, json, rss]
  formats:
    - html
    **- json**
